tests/e2e_test.go

```diff
@@ -282,6 +305,160 @@ func TestSignedEntryTimestamp(t *testing.T) {
 	}
 }
 
+func setup(t *testing.T) {
+	if err := os.MkdirAll("testdata", 0755); err != nil {
+		t.Fatalf("Failed to create testdata directory: %v", err)
+	}
+
+	if _, err := os.Stat(testCAPath); os.IsNotExist(err) {
+		t.Logf("Generating CA certificate: %s", testCAPath)
+		if err := generateTestCA(testCAPath, testCAKeyPath); err != nil {
+			t.Fatalf("Failed to generate CA certificate: %v", err)
+		}
+	}
+
+	if _, err := os.Stat(testSignerKey); os.IsNotExist(err) {
+		t.Logf("Generating signer key: %s", testSignerKey)
+		if err := generateTestSigner(testSignerKey, testSignerCert); err != nil {
+			t.Fatalf("Failed to generate signer key: %v", err)
+		}
+	}
+
+	viper.Set("rekor_server.signer", testSignerKey)
+	viper.Set("rekor_server.signer-passwd", testSignerPass)
+	viper.Set("trillian_log_server.tls", true)
+	viper.Set("trillian_log_server.tls_ca_cert", testCAPath)
+}
+
+func TestNewAPIWithTLS(t *testing.T) {
+	setup(t)
+
+	apiInstance, err := api.NewAPI(testTreeID)
+	if err != nil {
+		t.Fatalf("Failed to create API instance with TLS: %v", err)
+	}
+	t.Logf("API initialized with TLS: %+v", apiInstance)
+}
+
+func TestDialWithCustomCA(t *testing.T) {
+	setup(t)
+
+	tlsCACert, err := os.ReadFile(testCAPath)
+	if err != nil {
+		t.Fatalf("Failed to read CA certificate: %v", err)
+	}
+	certPool := x509.NewCertPool()
+	if !certPool.AppendCertsFromPEM(tlsCACert) {
+		t.Fatalf("Failed to append CA certificate to pool")
+	}
+
+	creds := credentials.NewTLS(&tls.Config{
+		RootCAs:    certPool,
+		MinVersion: tls.VersionTLS12,
+	})
+
+	conn, err := grpc.Dial(testServerAddr, grpc.WithTransportCredentials(creds))
```

Comment:
Rekor doesn't expose a gRPC endpoint.

Reply:
Ah never mind, you're connecting to Trillian. Can you update the variable name to reflect that?